src/main/java/io/github/fbiville/nestedrelationquerysdn6/Changeset.java

```diff
@@ -13,22 +16,34 @@
     @Property("USER")
     private String user;
 
-    @Relationship(type = "CHANGESET_IN")
-    private Map<Workspace, ChangesetIn> changesetIn = new HashMap<>();
+    @Relationship(direction = Direction.OUTGOING)
+    private Map<String, List<ChangesetIn>> changesetIn = new HashMap<>(); 
+    
+    @Relationship(type = "CHANGESET_IN", direction = Direction.OUTGOING) 
+    private List<Workspace> workspace = new ArrayList<>();
```

Comment:
I'm not sure I understand the need for both `changesetIn` and `workspace`.
If there are no other relationship types than `CHANGSET_IN` connecting changesets and workspaces, you only need the `workspace` field.
Otherwise, `changesetIn` will contain all the relationships going to a `Workspace` anyway and `ChangesetIn` contains a reference to it.

Reply:
You are completely right but I decided to let it for one reason. They query generated by data.Cypher when but properties are on the model is different than the one generated when there's just the changesetIn rel property. Check on that (remove workspace reference and some tests).